Added options for spawn, such as cwd.

ghost opened the pull request:
Thank you for the very helpful gulp-foreman plugin.  I found that I wanted to pass an options hash to the underlying spawn() call, in order to set a working directory other than the default gulp directory.  The approach I took was to pass through an options hash, which would allow for other options to the spawn call as well.

maldrake committed:
Added options for spawn, such as cwd.

jshcrowthe commented:
@maldrake This is awesome! I like the idea a lot. Before we merge this would you mind doing two things:
1. Make demo of this so others can see a sample usage
2. Update package.json (This'd be a feature addition so the new version would be `0.2.0`)

If you can go ahead and do those two things we'll verify and merge this one in.

maldrake committed:
Fixed bug on default config, added demo, and incremented version in package.json.

ghost commented:
Sounds good.  I added a demo for spawn-options, although the only one I set was the current working directly, as that was what I used it for.  Going through the demos also uncovered a bug I introduced.  I needed to move the handling of a null config hash (from a purely default invocation) out of buildArgs to cover options as well.  I also incremented to version 0.2.0.  Let me know if you spot anything else that needs fixing before the merge.
